sparkmagic/sparkmagic/kernels/wrapperkernel/usercodeparser.py

```diff
@@ -18,7 +18,7 @@ def get_code_to_run(self, code):
         except IndexError:
             all_but_first_line = ""
 
-        if code.startswith("%%local") or code.startswith("%local"):
+        if (code.startswith("%%local") or code.startswith("%local")) and not "-o" in code:
```

Comment:
This seems risky...

Reply:
Yes, I know this code smells, but so does the entire method(imo). I just followed the convention. I would have parsed users' first input line more sophisticatedly overall. Not sure if this PR is a good place to change it tho.

What would you suggest instead of this line?

Reply:
Yes... I've never been to happy with this class overall. How would you make it more sophisticated? I don't have suggestions as of now.